### PR TITLE
Resolves issues with Marking by Task Definition.

### PR DESCRIPTION
**Pull request by alexcu**
If I am a tutor and I have a group assigned to me, I should see all students from _any_ individual tutorial (i.e., not just my students in my tutorial).
    
E.g., Eda, Kattie and Michael are in a group. Eda is in Tutorial A, Katie is in Tutorial B, and Michael is in Tutorial C. I am a tutor for Tutorial C and a tutor for this group. I should therefore see Eda and Katie as well.

This PR resolves this issue, as previously I couldn’t see the other students.

### CODE REVIEW

File: src/app/common/services/project-service.coffee [hunk 1]
```diff
@@ -172,13 +172,13 @@ angular.module("doubtfire.common.services.projects", [])
...
       project.task_stats = updated_stats
-    
+
     project.updateBurndownChart = ->
       Project.get { id: project.project_id }, (response) ->
         project.burndown_chart_data = response.burndown_chart_data
         Visualisation.refreshAll()
 
-    project.incorporateTask = (newTask) ->
+    project.incorporateTask = (newTask, callback) ->
```
Comment (macite) [inline src/app/common/services/project-service.coffee]: Is the callback used here?

File: src/app/common/services/project-service.coffee [hunk 2]
```diff
@@ -214,14 +216,18 @@ angular.module("doubtfire.common.services.projects", [])
           onFailure?(failure)
       )
 
+  projectService.getGroup = (project, onSuccess) ->
+    Project.get { id: project.project_id }, (response) ->
+      project.groups = response.groups
+      onSuccess?(project)
+
   projectService.updateGroups = (project) ->
     if project.groups?
```
Comment (jakerenzella) [inline src/app/common/services/project-service.coffee]: Remove this if statement, and remove `getGroup` function.

File: src/app/common/services/project-service.coffee [hunk 3]
```diff
@@ -187,6 +187,8 @@ angular.module("doubtfire.common.services.projects", [])
       else
         project.tasks.push(projectService.mapTask(newTask, unit, project))
         currentTask = newTask
+      if currentTask.isGroupTask() and !currentTask.groups?
+        projectService.getGroup(currentTask.project(), callback)
       currentTask
```
Comment (jakerenzella) [inline src/app/common/services/project-service.coffee]: Else callback

File: src/app/common/services/project-service.coffee (hunk 2 above)
Comment (alexcu) [reply src/app/common/services/project-service.coffee]: 🥂

**Pull request merged**